cmd/cluster-cloud-controller-manager-operator/main.go

```diff
@@ -137,15 +145,62 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Setup for the feature gate accessor. This reads and monitors feature gates
+	// from the FeatureGate object status for the given version.
+	desiredVersion := controllers.GetReleaseVersion()
+	missingVersion := "0.0.1-snapshot"
+
+	configClient, err := configv1client.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		setupLog.Error(err, "unable to create config client")
+		os.Exit(1)
+	}
+	configInformers := configinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
+
+	kubeClient, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		setupLog.Error(err, "unable to create kube client")
+		os.Exit(1)
+	}
+
+	controllerRef, err := events.GetControllerReferenceForCurrentPod(ctx, kubeClient, *managedNamespace, nil)
+	if err != nil {
+		klog.Warningf("unable to get owner reference (falling back to namespace): %v", err)
+	}
+
+	recorder := events.NewKubeRecorder(kubeClient.CoreV1().Events(*managedNamespace), "cloud-controller-manager-operator", controllerRef)
+	featureGateAccessor := featuregates.NewFeatureGateAccess(
+		desiredVersion, missingVersion,
+		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
+		recorder,
+	)
+
+	featureGateAccessor.SetChangeHandler(func(featureChange featuregates.FeatureChange) {
```

Comment:
you sure it does?

Reply:
We've been watching and reacting to the feature gate changes since we started this operator, if it couldn't handle the change (eg from nothing to TPNU) then we would have hit issues before no?